### ISSUE TITLE
Prometheus query sometimes fails due to context deadline exceeded

### ISSUE THREAD

**Issue by mura-s** [pipe-cd/pipe#1245]
**What happened**:

When I set a `30s` interval in the `analysis-template.yaml` and a `60s` duration in the `.pipe.yaml`, Prometheus query **sometimes** fails due to context deadline exceeded.

```
[2020-12-14 12:57:32 +09:00]
[metrics-0] The query result is a failure. Reason: failed to run query: Post "http://service-prometheus.monitoring.svc.cluster.local:9090/api/v1/query": context deadline exceeded
[2020-12-14 12:57:32 +09:00]
Analysis failed: anslysis 'metrics-0' failed because the failure number exceeded the failure limit (0)
```

It's because `RunQuery()` in the Prometheus Provider returns an error when the duration of `ANALYSIS` step is exceeded and the parent context is timed out.

- https://github.com/pipe-cd/pipe/blob/master/pkg/app/piped/analysisprovider/metrics/prometheus/prometheus.go#L71-L74
- https://github.com/pipe-cd/pipe/blob/9e533a3d22e465f04fb1865b1fdd070f313f1282/pkg/app/piped/executor/analysis/analyzer.go#L53-L57
- https://github.com/pipe-cd/pipe/blob/9e533a3d22e465f04fb1865b1fdd070f313f1282/pkg/app/piped/executor/analysis/analysis.go#L109-L126

**What you expected to happen**:

In this case, `ANALYSIS` step succeeds.

**How to reproduce it**:

Set configurations like below, then execute a deployment.

```
# analysis-template.yaml
apiVersion: pipecd.dev/v1beta1
kind: AnalysisTemplate
spec:
  metrics:
    grpc_error_rate_percentage:
      interval: 30s
      provider: prometheus-dev
      expected:
        max: 10
      query: |
        100 - sum(
            rate(
                grpc_server_handled_total{
                  grpc_code!="OK",
                  kubernetes_namespace="{{ .K8s.Namespace }}",
                  kubernetes_pod_name=~"{{ .App.Name }}-[0-9a-zA-Z]+(-[0-9a-zA-Z]+)"
                }[{{ .Args.interval }}]
            )
        )
        /
        sum(
            rate(
                grpc_server_started_total{
                  kubernetes_namespace="{{ .K8s.Namespace }}",
                  kubernetes_pod_name=~"{{ .App.Name }}-[0-9a-zA-Z]+(-[0-9a-zA-Z]+)"
                }[{{ .Args.interval }}]
            )
        ) * 100
```

```
# .pipe.yaml
apiVersion: pipecd.dev/v1beta1
kind: KubernetesApp
spec:
  pipeline:
    stages:
      - name: ANALYSIS
        with:
          duration: 60s
          metrics:
            - template:
                name: grpc_error_rate_percentage
                args:
                  interval: 30s
```

**Environment**:
- `piped` version: v0.9.0
- `control-plane` version: v0.9.0

**Comment by nakabonne** [comment]
@mura-s As you feel, it shouldn't end with failure in that case. Hopefully we can identify the error type.

**Comment by nakabonne** [comment]
@mura-s Here it is! The fixed version has been released as [v0.9.7](https://github.com/pipe-cd/pipe/releases/tag/v0.9.7) 🎉  Please bump to it and try out it!
